### PR TITLE
Adding AgentDetails to be optional

### PR DESCRIPTION
**Pull request by bechampion**
Hi Guys.

This is sort of the proposal in relevance to the ticket:
https://github.com/jenkinsci/kubernetes-operator/issues/159

The idea is that the user could set the ServiceAccount that the builder will run on as:
```
apiVersion: jenkins.io/v1alpha2    
kind: Jenkins    
metadata:    
  name: sporting    
spec:    
  agent:    
    name: example    
    image: "jenkins/jnlp-slave:alpine"    
    serviceAccountName: example                                                                                                                                                                                      
  master:                                  
    basePlugins:                           

```
If not set , there's a number of constants that override  them details.

What you think?

### CODE REVIEW

File: pkg/apis/jenkins/v1alpha2/jenkins_types.go [hunk 1]
```diff
@@ -214,6 +218,15 @@ type Plugin struct {
...
 
+// Plugin defines Jenkins agent
+type Agent struct {
+	// Name is the name of Jenkins Agent
+	Name string `json:"name"`
+	// Image to be used for the Jenkins Agent
+	Image string `json:"image"`
+	// ServiceAccountName to be used on the Jenins Agent
+	ServiceAccountName string `json:"serviceAccountName"`
+}
```
Comment (pawelprazak) [inline pkg/apis/jenkins/v1alpha2/jenkins_types.go]: - [ ] the doc comment needs to be more clear, and needs to start with `Agent`
- [ ] we need a different structure, at least something like something like
```
Replicas Integer `json:"replicas,omitempty"`
Template []PodTemplateSpec `json:"template,omitempty"`
```
Probably some more and/or different structs (I'm going from the docs here)

File: pkg/apis/jenkins/v1alpha2/jenkins_types.go [hunk 2]
```diff
@@ -310,6 +323,9 @@ type JenkinsMaster struct {
 	// Plugins contains plugins required by user
 	// +optional
 	Plugins []Plugin `json:"plugins,omitempty"`
+
+        //BuildersServiceAccountName Name of the SA bound to the builder
+	BuilderServiceAccountName string `json:"builderServiceAccountName,omitempty"`
```
Comment (pawelprazak) [inline pkg/apis/jenkins/v1alpha2/jenkins_types.go]: I don't think we need this here.

File: pkg/controller/jenkins/configuration/user/seedjobs/seedjobs.go [hunk 3]
```diff
@@ -42,6 +42,12 @@ const (
 	// AgentName is the name of seed job agent
 	AgentName = "seed-job-agent"
 
+	// AgentImage is the Image the agent will use
+	AgentImage = "jenkins/jnlp-slave:alpine"
+
+	// AgentServiceAccountName is the SA the agent will run as
+	AgentServiceAccountName = "default"
+
```
Comment (pawelprazak) [inline pkg/controller/jenkins/configuration/user/seedjobs/seedjobs.go]: Seed job agent definition would have to be refactored to use the new API and would be the default Agent in `[]Agents`

What do you think @tomaszsek?

**Comment by bechampion** [comment]
Thanks for all these comments ill fix what I can soon and get back to you, thank you again for looking into this

**Comment by pawelprazak** [comment]
closing in favour of #196

**Pull request closed without merging**